### PR TITLE
[CARBONDATA-3167] Add a example for DataFrame read/write non-transactional table from/to S3

### PR DESCRIPTION
**Pull request by xubo245**
Add a example for DataFrame read/write non-transactional table from/to S3

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 No
 - [x] Any backward compatibility impacted?
 No
 - [x] Document update required?
No
 - [x] Testing done
     add test case
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 

No

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1742/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1953/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10001/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1743/

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/DataFrameSDKS3Example.scala [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+/**
+  * This example show DataFrame How to read/Write SDK data from/to S3
+  */
+object DataFrameSDKS3Example {
+
+  val rootPath = new File(this.getClass.getResource("/").getPath
+    + "../../../..").getCanonicalPath
+
+  def main(args: Array[String]): Unit = {
```
[the remark this answers is not in the document]
Comment (xubo245) [reply examples/spark2/src/main/scala/org/apache/carbondata/examples/DataFrameSDKS3Example.scala]: This is a example, please check other example, like: org.apache.carbondata.examples.CarbonSessionExample#main

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/DataFrameSDKS3Example.scala [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+    try {
+      val df = sparkSession.emptyDataFrame
+
+      var path = s"$rootPath/examples/spark2/target/carbon"
+      if (args.length == 1) {
+        path = args(0)
+      }
+      if (args.length == 3) {
+        path = "s3a://xubo/sdk/DFTest"
```
[the remark this answers is not in the document]
Comment (xubo245) [reply examples/spark2/src/main/scala/org/apache/carbondata/examples/DataFrameSDKS3Example.scala]: done.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1744/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10003/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1955/

**Comment by xubo245** [comment]
@jackylk @QiangCai @KanakaKumar @kunal642 Please review it.

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/DataFrameSDKS3Example.scala [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        StructField("id", StringType, nullable = false),
+        StructField("m1", ShortType, nullable = false),
+        StructField("m2", IntegerType, nullable = false),
+        StructField("m3", LongType, nullable = false),
+        StructField("m4", DoubleType, nullable = false),
+        StructField("m5", DoubleType, nullable = false)
+      )
+    )
+
+    carbonSession.createDataFrame(rdd, schema)
```
Comment (BJangir) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/DataFrameSDKS3Example.scala]: @xubo245 ,Please try rdd.toDF directly and pass columns name in selectExpr . For this you need to import carbonsession.implicits._  .

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1908/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10162/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2119/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10218/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2239/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/42/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/41/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.4/43/

**Pull request closed without merging**